### PR TITLE
New Feature: overload of GetSetting() that requires setting comes from ServiceRuntime, or else throws

### PR DESCRIPTION
**Pull request by MichaelCommo**
Added overload of GetSetting() (in CloudConfigurationManager and AzureApplicationSetting classes) that adds support for throwing exception if setting is not found in the ServiceRuntime.  Existing GetSetting() methods are unchanged to ensure backwards-compatibility.  Reason for this feature: falling back on local file-based configuration could actually mean missing/incomplete configuration, and the consumer of this package will now be able to disallow that from happening.

Can we also please get a new version of Nuget package: Microsoft.WindowsAzure.ConfigurationManager published to include this change?

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@MichaelCommo__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@MichaelCommo__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by MichaelCommo** [comment]
CLA has been signed by all all parties on our side. Thanks!
https://app.docusign.com/documents/details/ffec5d5d-3b52-4601-8e05-e650a21467b8

**Comment by MichaelCommo** [comment]
Note that I did not change/increment the assembly version in AssemblyInfo.cs in this changeset; but I would greatly appreciate having the version incremented to 3.2.2 and a new nuget package published that includes this change.
`[assembly: AssemblyFileVersion("3.2.2.0")]`

**Comment by hovsepm** [comment]
@azuresdkci test this please

**Comment by hovsepm** [comment]
@MichaelCommo please increment package version in AssemlbyInfo.cs (AssemlbyFileVersion as you mentioned in your comment above), and in nuget.proj file - https://github.com/Azure/azure-sdk-for-net/blob/master/src/Common/Configuration/Microsoft.WindowsAzure.ConfigurationManager.nuget.proj

**Comment by MichaelCommo** [comment]
@hovsepm should I make those two file changes you requested a separate pull request from this one?  Or is there a way for me to add those changes to this pull request?

**Comment by MichaelCommo** [comment]
@hovsepm I've incremented the package version in the places you requested. Thank you for the help!

File: src/Common/Configuration/AzureApplicationSettings.cs [hunk 1]
```diff
@@ -111,6 +111,34 @@ private bool IsMissingSettingException(Exception e)
...
+            Debug.Assert(!string.IsNullOrEmpty(name));
+
+            string value = null;
+
+            value = GetValue("ServiceRuntime", name, GetServiceRuntimeSetting, outputResultsToTrace);
+            if (value == null)
+            {
+                if (throwIfNotFoundInRuntime)
+                {
+                    throw new SettingsPropertyNotFoundException($"Setting: {name} was not found in the ServiceRuntime.");
```
Comment (hovsepm) [inline src/Common/Configuration/AzureApplicationSettings.cs]: Seems this string should go to module resources. Please fix

**Review by hovsepm — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@hovsepm can you sign off on this?

**Review by hovsepm — APPROVED** [review]

File: src/Common/Configuration/Properties/Resources.Designer.cs [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-//     Runtime Version:4.0.30319.17929
+//     Runtime Version:4.0.30319.42000
 //
 //     Changes to this file may cause incorrect behavior and will be lost if
 //     the code is regenerated.
 // </auto-generated>
 //------------------------------------------------------------------------------
 
-namespace Microsoft.WindowsAzure {
+namespace Microsoft.WindowsAzure.Properties {
```
Comment (hovsepm) [inline src/Common/Configuration/Properties/Resources.Designer.cs]: @MichaelCommo please revert this change.

File: src/Common/Configuration/Properties/Resources.Designer.cs [hunk 3]
```diff
@@ -39,7 +39,7 @@ internal class Resources {
         internal static global::System.Resources.ResourceManager ResourceManager {
             get {
                 if (object.ReferenceEquals(resourceMan, null)) {
-                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.WindowsAzure.Resources", typeof(Resources).Assembly);
+                    global::System.Resources.ResourceManager temp = new global::System.Resources.ResourceManager("Microsoft.WindowsAzure.Properties.Resources", typeof(Resources).Assembly);
```
Comment (hovsepm) [inline src/Common/Configuration/Properties/Resources.Designer.cs]: and this change as well

**Review by hovsepm — CHANGES_REQUESTED** [review]

File: src/Common/Configuration/Properties/Resources.Designer.cs (hunk 3 above)
Comment (MichaelCommo) [inline src/Common/Configuration/Properties/Resources.Designer.cs]: Absolutely, I have just rolled back these changes. Just a side note: as the resources designer is generated code, and as the namespace tends to follow project structure, any changes to the resources file will automatically result in this namespace change in the future.  Thanks again for your continued support for this new feature.

Comment (hovsepm) [inline src/Common/Configuration/Properties/Resources.Designer.cs]: This one is still in your PR. Please make sure you pushed your changes to your fork.

**Review by hovsepm — CHANGES_REQUESTED** [review]

**Comment by MichaelCommo** [comment]
@hovsepm: Should be fixed.

**Comment by MichaelCommo** [comment]
@hosepm,


The commit shows up in the commits tied to the pull request: https://github.com/Azure/azure-sdk-for-net/pull/2459


I'm pretty sure the change is part of this pull request, as I submitted it as an "append to previous commit".  You can see the commit in the "commits" section.  I think that the reason you do not see the the change in the "files changed" section is because it's no longer a change (its the same as the azure master branch now).


Thanks,

Mike


________________________________

**Comment by hovsepm** [comment]
@MichaelCommo yeah now it is fixed :) thank you for the contribution!

**Comment by hovsepm** [comment]
00:07:56.944   Pushing Microsoft.WindowsAzure.ConfigurationManager.3.2.2.nupkg to 'https://www.nuget.org/api/v2/package/'...
00:07:56.965     PUT https://www.nuget.org/api/v2/package/
00:08:05.293     Created https://www.nuget.org/api/v2/package/ 8324ms
00:08:05.293   Your package was pushed.

**Comment by hovsepm** [comment]
@MichaelCommo  you should be able to pull the new version

**Comment by MichaelCommo** [comment]
@hovsepm: Thank you! Much appreciated. Pulling down the package as we speak!
